src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -214,113 +213,120 @@ export default class Blockchain extends Emittery.Typed<
       }
     }
 
-    const database = (this.#database = new Database(options.database, this));
-    database.once("ready").then(async () => {
-      const blocks = (this.blocks = await BlockManager.initialize(
-        common,
-        database.blockIndexes,
-        database.blocks
-      ));
+    this.coinbase = coinbaseAddress;
 
-      // if we have a latest block, use it to set up the trie.
-      const latest = blocks.latest;
-      if (latest) {
-        this.#blockBeingSavedPromise = Promise.resolve({
-          block: latest,
-          blockLogs: null
-        });
-        this.trie = new SecureTrie(
-          database.trie,
-          latest.header.stateRoot.toBuffer()
-        );
-      } else {
-        this.trie = new SecureTrie(database.trie, null);
-      }
-
-      this.blockLogs = new BlockLogManager(database.blockLogs);
-      this.transactions = new TransactionManager(
-        options.miner,
-        common,
-        this,
-        database.transactions
-      );
-      this.transactionReceipts = new Manager(
-        database.transactionReceipts,
-        TransactionReceipt
-      );
-      this.accounts = new AccountManager(this, database.trie);
-      this.storageKeys = database.storageKeys;
+    this.#database = new Database(options.database, this);
+  }
 
-      this.coinbase = coinbaseAddress;
+  async initialize(initialAccounts: Account[]) {
+    await this.#database.initialize();
 
-      // create VM and listen to step events
-      this.vm = this.createVmFromStateTrie(
-        this.trie,
-        options.chain.allowUnlimitedContractSize
+    const database = this.#database;
```

Comment:
Since we're assigning to `this.#database` to `database`, let's use the `database` variable in the `await this.#database.initialize();` line above (line `222`).